awx/main/credential_plugins/aim.py

```diff
@@ -0,0 +1,83 @@
+from .plugin import CredentialPlugin
+
+import os
+import stat
+import tempfile
+import threading
+from urllib.parse import urljoin
+
+from django.utils.translation import ugettext_lazy as _
+import requests
+
+
+aim_inputs = {
+    'fields': [{
+        'id': 'url',
+        'label': _('AIM URL'),
+        'type': 'string',
+    }, {
+        'id': 'application_id',
+        'label': _('Application ID'),
+        'type': 'string',
+        'secret': True,
+    }, {
+        'id': 'client_key',
+        'label': _('Client Key'),
+        'type': 'string',
+        'secret': True,
+        'multiline': True,
+    }, {
+        'id': 'client_cert',
+        'label': _('Client Cert'),
+        'type': 'string',
+        'secret': True,
+        'multiline': True,  
+    }],
+    'metadata': [{
+        'id': 'safe',
+        'label': _('Safe'),
+        'type': 'string',
+    }, {
+        'id': 'object',
+        'label': _('Object'),
+        'type': 'string',
+    }],
+    'required': ['url', 'application_id', 'safe', 'object'],
+}
+
+def temp_fifo(data):
+    # Open fifo named pipe in a new thread.
```

Comment:
Certificate data is stored in the db, we're using a fifo here so we can avoid writing it to disk in order to make the request.

We probably don't want to be opening threads if we can avoid it, so I'm currently looking into writing a custom transport adapter that would read the cert data from a string instead of expecting a file path http://docs.python-requests.org/en/master/user/advanced/#transport-adapters

Reply:
After doing some digging, I think this is a deeper problem than what can be solved with a custom transport adapter. You _can_ create and configure your own ssl context object with this approach, but the ssl context api itself [will still require a file path](https://bugs.python.org/issue16487).

Some basic testing has shown the fifo work, and we've used this approach in other places so the current plan is to use it here.